src/Education/XamlGenerator/CustomTextMarkupParser.cs

```diff
@@ -42,15 +42,20 @@ namespace SonarLint.VisualStudio.Education.XamlGenerator
     /// </summary>
     internal static class CustomTextMarkupParser
     {
+        /// <summary>
+        /// Matches - ... text ... {rule:cpp:S123} ... text ... {rule:cpp:S999} ... text
+        /// </summary>
+        private static readonly Regex RuleCrossRefRegEx = new Regex("{rule:(?<repoKey>[A-Za-z]+):(?<ruleKey>[\\w]+)}",
+            RegexOptions.Compiled, RegexConstants.DefaultTimeout);
+
         /// <summary>
         /// Parse a text string that might/might not contain rule cross-references
         /// </summary>
         public static IEnumerable<ITextToken> Parse(string text)
         {
             var textTokens = new List<ITextToken>();
 
-            // Matches -   ... text ... {rule:cpp:S123} ... text ... {rule:cpp:S999} ... text
-            var matches = Regex.Matches(text, "{rule:(?<repoKey>[A-Za-z]+):(?<ruleKey>[\\w]+)}");
```

Comment:
Seeing the hotspot warning in your PR for this class was actually what prompted me to create the ticket about fixing all of the RegExs - but I must have created my branch before your PR was merged.